src/server_manager/web_app/app.spec.ts

```diff
@@ -35,41 +32,43 @@ const TOKEN_WITH_ONE_SERVER = 'one-server-token';
 // tslint:disable-next-line:no-any
 (global as any).bringToFront = () => {};
 
+// Inject app-root element into DOM once before the test suite runs
+beforeAll(async () => {
+  const loadAppRoot = new AppRoot();  // TODO: Remove once figure out how to load AppRoot properly.
+
+  const appRootEl = document.createElement('app-root');
+  appRootEl.setAttribute('id', 'appRoot');
+  document.getElementsByTagName('body')[0].appendChild(appRootEl);
```

Comment:
Does document have a body? Where's the document defined?

Reply:
Karma creates and empty iframe with no document body, so I manually create it here.

My understanding is that Karma is just the test runner and that manipulating the DOM is outside of its responsibilities. We can introduce other testing libraries if we don't want to do it manually, but I figured that we first unblock the app tests and add other tools as needed.

It would be nice to have a tool that loads the web component being tested.